src/Button.php

```diff
@@ -12,81 +13,206 @@
-    private bool $encodeTags = false;
-    private array $options = [];
+    private string $tagName = 'button';
 
-    public function render(): string
+    /**
+     * Sets the button to be active.
+     *
+     * @return self A new instance of the current class with the button active.
+     */
+    public function active(): self
     {
-        if (!isset($this->options['id'])) {
-            $this->options['id'] = $this->getId();
-        }
+        $new = clone $this;
+        $new->attributes['aria-pressed'] = 'true';
+        $new->attributes['data-bs-toggle'] = 'button';
+        $new->cssClass = 'active';
 
-        if ($this->theme) {
-            $this->options['data-bs-theme'] = $this->theme;
-        }
+        return $new;
+    }
 
-        /** @psalm-suppress InvalidArgument */
-        Html::addCssClass($this->options, ['widget' => 'btn']);
+    /**
+     * Sets the CSS class attribute for the button component.
+     *
+     * @param string $value The CSS class for the button component (e.g., 'test-class').
+     *
+     * @return self A new instance of the current class with the specified class value.
+     *
+     * @link https://html.spec.whatwg.org/#classes
+     */
+    public function addClass(string $value): self
+    {
+        $new = clone $this;
+        $new->cssClass = $value;
 
-        return Html::tag(
-            $this->tagName,
-            $this->encodeLabels ? Html::encode($this->label) : $this->label,
-            $this->options
-        )
-            ->encode($this->encodeTags)
-            ->render();
+        return $new;
     }
 
     /**
-     * When tags Labels HTML should not be encoded.
+     * Sets the HTML attributes for the button component.
+     *
+     * @param array $values Attribute values indexed by attribute names.
+     *
+     * @return self A new instance of the current class with the specified attributes.
+     *
+     * @see {\Yiisoft\Html\Html::renderTagAttributes()} for details on how attributes are being rendered.
      */
-    public function withoutEncodeLabels(): self
+    public function attributes(array $values): self
     {
         $new = clone $this;
-        $new->encodeLabels = false;
+        $new->attributes = $values;
 
         return $new;
     }
 
     /**
-     * The button label
+     * Sets the 'data-bs-toggle' attribute for the button.
+     *
+     * @param string $value The value to set for the 'data-bs-toggle' attribute.
+     *
+     * @return self A new instance of the current class with the specified 'data-bs-toggle' value.
      */
-    public function label(string $value): self
+    public function dataBsToggle(string $value): self
     {
+        $new = clone $this;
+        $new->attributes['data-bs-toggle'] = $value;
+
+        return $new;
+    }
+
+    /**
+     * Sets the button to be disabled.
+     *
+     * @return self A new instance of the current class with the button disabled.
+     */
+    public function disabled(): self
```

Comment:
How about passing a boolean here?


```suggestion
    public function disabled(bool $value = true): self
```

In case of `false` we may remove the attribute.

Reply:
done.